### DISCUSSION TITLE
Move v2/testsuite to v2/internal/testsuite

### DISCUSSION THREAD

**Discussion by pelletier** [pelletier/go-toml#728] [General]
@moorereason I'm looking at the exposed API of v2 in preparation for the stable release (finally!). I saw the package `v2/testsuite` is public. I only see two uses of it:

In `gotoml-test-decoder`:

https://github.com/pelletier/go-toml/blob/4807229e94f94f6299b4a5ca3d4ad45d1fe35907/cmd/gotoml-test-decoder/main.go#L9

And `toml_testgen_support_test.go`:

https://github.com/pelletier/go-toml/blob/4807229e94f94f6299b4a5ca3d4ad45d1fe35907/toml_testgen_support_test.go#L11

Do you foresee using those functions outside of go-toml or do you mind if I move them to `v2/internal/testsuite` to avoid making them part of the public API?

**Comment by moorereason** [comment]
I agree.  Let's move it to internal.

If my memory serves me, I was using the API initially to do direct Go differential fuzzing against BurntSushi.  I'm now fuzzing the CLI decoders so that I can fuzz against non-Go implementations.

**Comment by moorereason** [comment]
BTW, we also need to run `go generate` to regen the tests from the `toml-test` suite.
